integrations/firehydrant/client.py

```diff
@@ -0,0 +1,451 @@
+
+        Args:
+            report (dict[str, Any]): The incident report data to be enriched.
+
+        Returns:
+            dict[str, Any]: The enriched incident report with added information.
+        """
+
+        ## get incident task completion summary statistics
+        incident_id = report.get("incident", {}).get("id")
+        task_completion_status = await self.get_tasks_by_incident(
+            incident_id=incident_id
+        )
+
+        ## get incident duration data
+        milestones = report.get("incident", {}).get("milestones", [])
+        incident_duration = await self.compute_incident_duration(milestones=milestones)
+
+        ## get retrospective questions and answers
+        retro_responses = await self.get_retrospective_questions_and_answers(
+            questions=report.get("questions", [])
+        )
+
+        return {
+            "tasks": task_completion_status,
+            "duration": incident_duration,
+            "questions": retro_responses,
+        }
+
+    async def get_tasks_by_incident(self, incident_id: str) -> dict[str, Any]:
+        """
+        Retrieve tasks associated with a specific incident. The returned data includes details about each task, such as the number of completed and incompleted tasks.
+
+        Args:
+            incident_id (str): The unique identifier of the incident for which to fetch tasks.
+
+        Returns:
+            dict[str, Any]: A dictionary containing information about tasks related to the incident.
+        """
+        logger.info(f"Getting tasks details for incident: {incident_id}")
+        params: dict[str, Any] = {
+            "page": 1,
+            "per_page": PAGE_SIZE,
+        }
+        task_url = f"incidents/{incident_id}/tasks"
+
+        tasks = await self.send_paginated_api_request(
+            endpoint=task_url, query_params=params, data_key="data"
+        )
+        task_completion_status = await self.get_task_completion_status(tasks)
+        return task_completion_status
+
+    async def get_task_completion_status(
+        self, tasks: list[dict[str, Any]]
+    ) -> dict[str, Any]:
+        """
+        Return statistics about completed and uncompleted tasks. It returns a dictionary with counts of completed and uncompleted tasks, providing insights into task progress.
+
+        Args:
+            tasks (list[dict[str, Any]]): A list of task objects, each represented as a dictionary.
+
+        Returns:
+            dict[str, Any]: A dictionary containing statistics about completed and uncompleted tasks.
+        """
+        logger.info("Computing task completion statistics")
+        total_completed_tasks = 0
+        for task in tasks:
+            if task.get("state") == "done":
+                total_completed_tasks += 1
+
+        return {
+            "completed": total_completed_tasks,
+            "incompleted": len(tasks) - total_completed_tasks,
+        }
```

Comment:
i think this can be done by jq in the port app config

Reply:
calculation moved to jq